### PR TITLE
Hotfix: Fix broken serde

### PR DESCRIPTION
**Pull request by sanket1729**
My local scripts did not test serde feature on the merge commit. While merging #734, I accidentally broke the serde feature on the latest master. This PR fixes it.

Sorry about that, I will update my local script to be the same as of CI check.

### CODE REVIEW

**Review by tcharding — APPROVED** [review]
ACK 5fc40baa7388f43ea524557c8f5a0bfa555c1c36

**Review by Kixunil — APPROVED** [review]
ACK 5fc40baa7388f43ea524557c8f5a0bfa555c1c36

File: bitcoin/tests/serde.rs [hunk 1]
```diff
@@ -296,7 +296,7 @@ fn serde_regression_psbt() {
                 "304402204f67e2afb76142d44fae58a2495d33a3419daa26cd0db8d04f3452b63289ac0f022010762a9fb67e94cc5cad9026f6dc99ff7f070f4278d30fbc7d0c869dd38c7fe701".parse().unwrap(),
             )].into_iter().collect(),
             bip32_derivation: keypaths.clone().into_iter().collect(),
-            final_script_witness: Some(Witness::from_vec(vec![vec![1, 3], vec![5]])),
+            final_script_witness: Some(Witness::from_slice(&[vec![1, 3], vec![5]])),
```
Comment (Kixunil) [inline bitcoin/tests/serde.rs:299]: Nit: the inner vecs could've been changed to slices but let's keep it for some bored students. :)

**Comment by apoelstra** [comment]
Wow, fact action! Broken, fixed, two acks and merged, all while I was asleep.

Yes, figuring out bors is on my TODO list (though would bors have caught this? I don't know) ... though FWIW I now test all feature combination on merge commits locally.

**Comment by Kixunil** [comment]
> would bors have caught this?

If we used it exclusively, yes.

**Pull request merged**